### ISSUE TITLE
Show better log in errors

### ISSUE THREAD

**Issue by jzongker** [LiveChurchSolutions/AppBase#144]
I really thought we used to have friendly "invalid email or password" type errors, but we seem to just get this generic error anytime a login fails now.  Need to track this down and replace with a friendly error.

![image](https://user-images.githubusercontent.com/1447203/169385464-93b5892e-e059-424f-9de7-48efb6e6a36f.png)

**Comment by PhotoPaul** [comment]
I suspect this has to do with how the server is setup to respond.

When authorization fails at the localhost it responds with this:
![image](https://user-images.githubusercontent.com/3471804/169496812-3a8bfac3-8acc-4387-9e86-97151ab7e814.png)

But when authorization fails at the staging server it responds with this:
![image](https://user-images.githubusercontent.com/3471804/169497034-557409d2-e6a3-4c55-9e8f-ba3411c750db.png)

For this kind of web apps I'm usually against using status codes other than 200 OK.
I suggest we consider replacing all 401s with 200, and if there's an error we handle it at web app level.

**Comment by jzongker** [comment]
I don't think the different 401 status code is the cause in this case.  AWS is probably just sticking to the strict HTTP standard where Node is not.  We don't read that status code message but rather the body which is being returned correctly even in this case.
![image](https://user-images.githubusercontent.com/1447203/169555924-3004a500-716c-47cc-ba7f-61856b8ac9c4.png)

We try not to ever return a 500 error message, but nearly every route across all the APIs returns a 401 if you're not authorized to call it.  It would mean updating hundreds of (maybe a thousand) calls to change at this point.  There are client-side checks as well to prevent you from making those API calls in the first place, but have to back those up with server-side checks too.

**Comment by PhotoPaul** [comment]
I looked further into it and I really don't see many options.

Here is the server side and the client side:

### ApiBase: [CustomBaseController.ts#L23-L25](https://github.com/LiveChurchSolutions/ApiBase/blob/cb0295abe1b2980c03c7d1a380f3cf873ab19da2/controllers/CustomBaseController.ts#L23-L25)

```
    public denyAccess(errors: string[]) {
        return this.json({ errors }, 401);
    }
```

### AppBase: [ApiHelper.ts#L135-L144](https://github.com/LiveChurchSolutions/AppBase/blob/2f1db98fbe6154bd7c710567eebb74b96ad33602/helpers/ApiHelper.ts#L135-L144)
```
    try {
      const response = await fetch(config.url + path, requestOptions);
      if (!response.ok) {
        throw new Error(response.statusText);
      }
      return response.json();
    } catch (e) {
      console.log(e)
      throw (e);
    }
```

If the server returns 401, then the response.ok is false and the client throws an Error forcing execution into the catch block and never returning the response.json()

So either we change the 401 to 200 on `CustomBaseController.ts#L24`, or we delete lines `ApiHelper.ts#137-139`

**Comment by jzongker** [comment]
I made two changes I think will resolve this issue and a fair number of future issues.  The problem is that the error details are being returned as part of the 401 body instead of the status text, so we just need to read the body to get the error details.  The first change I made was to do that for all API calls.

The second is, in the case of the login error, we really don't necessarily care what details are returned by the API.  All we need to do is show a prettier error message to the user to let them know their login failed, regardless of the details.  There was existing code in place to do this that was broken in Feb with #133.  That's fixed now.

https://github.com/LiveChurchSolutions/AppBase/commit/3d5f8f8375736dc636880b378c5b7da103bb4868
